app/src/main/java/clarkson/ee408/tictactoev4/PairingActivity.java

```diff
@@ -134,44 +254,123 @@ private void createRespondAlertDialog(Event invitation) {
+            try {
+                Response response = SocketClient.getInstance()
+                        .sendRequest(request, Response.class);
+
+                if (response == null) {
+                    AppExecutors.getInstance().mainThread().execute(() ->
+                            Toast.makeText(this, "Accept invitation failed.", Toast.LENGTH_SHORT).show()
+                    );
+                    return;
+                }
+
+                if (response.getStatus() == ResponseStatus.SUCCESS) {
+                    AppExecutors.getInstance().mainThread().execute(() ->
+                            handleAcceptInvitation(invitation)
+                    );
+                } else {
+                    AppExecutors.getInstance().mainThread().execute(() ->
+                            Toast.makeText(this, response.getMessage(), Toast.LENGTH_SHORT).show()
+                    );
+                }
+
+            } catch (Exception e) {
+                Log.e(TAG, "Error accepting invitation", e);
+            }
+        });
     }
 
+
     /**
      * Sends an DECLINE_INVITATION to the server
      * @param invitation the Event invitation to decline
      */
     private void declineInvitation(Event invitation) {
-        // TODO:  Send a DECLINE_INVITATION request to the server. If SUCCESS response, Toast a message, else, Toast the error
 
-        // TODO: set shouldUpdatePairing to true after DECLINE_INVITATION is sent.
+        if (invitation == null) {
+            Toast.makeText(this, "Invalid invitation.", Toast.LENGTH_SHORT).show();
+            return;
+        }
+
+        Request request = new Request();
+        request.setType(RequestType.DECLINE_INVITATION);
+        request.setData(String.valueOf(invitation.getEventId()));
+
+        AppExecutors.getInstance().networkIO().execute(() -> {
+            try {
+                Response response = SocketClient.getInstance()
+                        .sendRequest(request, Response.class);
+
+                if (response == null) {
+                    AppExecutors.getInstance().mainThread().execute(() ->
+                            Toast.makeText(this, "Decline invitation failed.", Toast.LENGTH_SHORT).show()
+                    );
+                    return;
+                }
+
+                AppExecutors.getInstance().mainThread().execute(() -> {
+                    if (response.getStatus() == ResponseStatus.SUCCESS) {
+                        Toast.makeText(this, "Invitation declined.", Toast.LENGTH_SHORT).show();
+                        shouldUpdatePairing = true;   // required by TODO
+                    } else {
+                        Toast.makeText(this, response.getMessage(), Toast.LENGTH_SHORT).show();
+                    }
+                });
+
+            } catch (Exception e) {
+                Log.e(TAG, "Error declining invitation", e);
+            }
+        });
     }
 
+
+
     /**
      *
```

Comment:
Javadoc needs a description

```suggestion
     * Begin the game in MainActivity
     * 
```